lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -3467,6 +3299,94 @@ LogicalResult FIRRTLLowering::visitStmt(SkipOp op) {
   return success();
 }
 
+void FIRRTLLowering::lowerConnect(Operation *regOp, Value dest, Value srcVal) {
+  // Unroll all accesses since they must be traversed in inverse order.
+  SmallVector<Operation *> accesses;
+  while (dest) {
+    TypeSwitch<Operation *, void>(dest.getDefiningOp())
+        .Case<SubfieldOp, SubindexOp>([&](auto op) {
+          accesses.push_back(op);
+          dest = op.input();
+        })
+        .Default([&](auto op) {
+          assert(op == regOp && "invalid lowering");
```

Comment:
I'm trying to grok this... for a reg, this is totally trivial, but for subfield and subindex, we do this unrolling?

I might put a .Case<RegOp, RegResetOp> with this assertion to make that a little more clear.

Also, is this unrolling something new we have to do to be able to support the new reg op? If it's separable, I would try to separate it to another PR.

Reply:
This should be moved into `PrepareForEmission` at some point, however for now I'd like to keep them together since the expansion of injects/concats is strictly limited to what is being generated here. We can then later think of a generic transformation that applies to all wires, considering how it affects readability and simulation performance. For now, I wish this to be tied with the register lowering to minimise differences to the baseline.

Reply:
So this is about keeping the test churn small? Can you add this first?

Reply:
If previously we had `connect reg.x, y`, in SV we would emit `reg.x <= y`. If we convert the connect into `x = hw.struct.inject reg["x"], y`, we would emit `reg = { x: y }`. We don't use injects anywhere yet and now register lowering should figure out how to do this properly.

Reply:
Would it work if I added FIR registers in one PR and the changed to `LowerToHW`/`LowerSeqToSV` in another?

Reply:
Separated work into #3190 and #3191

Reply:
Awesome, that makes sense, and thanks for splitting it out.